sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/misc.scala

```diff
@@ -66,11 +66,11 @@ case class PrintToStderr(child: Expression) extends UnaryExpression {
   """,
   since = "3.1.0",
   group = "misc_funcs")
-case class RaiseError(child: Expression) extends UnaryExpression with ImplicitCastInputTypes {
+case class RaiseError private[spark] (child: Expression, dataType: DataType)
```

Comment:
catalyst is already a private module. We can just remove `private[spark]`.

Reply:
+1